numpy/core/tests/test_multiarray.py

```diff
@@ -6116,6 +6116,27 @@ def test_array_interface_itemsize():
     assert_equal(descr_t.itemsize, typestr_t.itemsize)
 
 
+def test_array_interface_empty_shape():
+    arr = np.array([1, 2, 3])
+    interface1 = dict(arr.__array_interface__)
+    interface1['shape'] = ()
+
+    class DummyArray1(object):
+        __array_interface__ = interface1
+
+    interface2 = dict(interface1)
+    interface2['data'] = arr[0].tostring()
```

Comment:
Plus, I think the `tobytes` method looks better, not sure if we are deprecating the tostring one or not.